The password optional feature documentation - main publication PR. (#3605)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/react/integrationsteps.md

### 1: The user navigates to the sign-in page

When the user navigates to your app's sign-in page, call `OktaAuth.idx.start()` to start the sign-in flow.

```javascript
const { oktaAuth } = useOktaAuth();
const startFlow = useCallback(async () => {
  newTransaction = await oktaAuth.idx.start();
}, [oktaAuth, flow, setTransaction]);
```

> The method returns an `IdxTransaction` object containing field metadata that can be used to create a sign-in page dynamically.

Reviewer: ```suggestion
The method returns an `IdxTransaction` object that contains field metadata that you can use to create a sign-in page dynamically.
```

Author: updated.

<!--
Source image: https://www.figma.com/file/YH5Zhzp66kGCglrXQUag2E/%F0%9F%93%8A-Updated-Diagrams-for-Dev-Docs?node-id=2842%3A4925 pwd-optional-react-sign-up-verify-otp
 -->

</div>

When the user submits the OTP, create an object with a `verificationCode` property set to the OTP entered by the user.

```json
const inputValues = {
    "verificationCode":"197277"
 };
```

Call `OktaAuth.idx.proceed()` passing in the new object as a parameter.

```javascript
    const newTransaction = await oktaAuth.idx.proceed(inputValues);
    setTransaction(newTransaction);
    setInputValues({});
    setProcessing(false);
```

### 5. Identity Engine verifies OTP and returns success

`OktaAuth.idx.proceed()` returns `IdxTransaction.status` equal to `SUCCESS` along with access and ID tokens, which indicates a successful user sign-in flow.

```json
{
  status: "SUCCESS",
  tokens: {
    accessToken: {
      accessToken: "eyJraWQiOiJLSWdvVHlt...",
      expiresAt: 1656106249,
      tokenType: "Bearer",
    },
    idToken: {
      idToken: "eyJraWQiOiJLSWdvVHltSGlL...",
      expiresAt: 1656106249,
    },
  },
}
```

> Store these tokens for future requests and redirect the user to the default page after a successful sign-up attempt.

Author: updated.
